recipes/enjincppsdk/all/conanfile.py

```diff
@@ -0,0 +1,139 @@
+            "Visual Studio": "16",
+            "gcc": "9",
+            "clang": "10",
+        }
+
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+        if self.options.with_default_http_client:
+            self.options["cpp-httplib"].with_openssl = True
+
+        self.options["spdlog"].header_only = True
+
+    def requirements(self):
+        if self.options.with_default_http_client:
+            self.requires("cpp-httplib/0.8.5")
+
+        if self.options.with_default_ws_client:
+            self.requires("ixwebsocket/11.0.4")
+
+        self.requires("rapidjson/1.1.0")
+        self.requires("spdlog/1.8.2")
+
+    def build_requirements(self):
+        self.build_requires("cmake/3.16.9")
+
+    def validate(self):
+        # Validations for OS
+        if self.settings.os == "Macos":
+            raise ConanInvalidConfiguration("macOS is not supported at this time. Contributions are welcomed.")
+
+        # Validations for minimum required C++ standard
+        compiler = self.settings.compiler
+
+        if compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, 17)
+
+        minimum_version = self._minimum_compilers_version.get(str(compiler), False)
+        if not minimum_version:
+            self.output.warn("C++17 support is required. Your compiler is unknown. Assuming it supports C++17.")
+        elif tools.Version(compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration("C++17 support is required, which your compiler does not support.")
+
+        if compiler == "clang" and compiler.libcxx != "libstdc++11":
+            raise ConanInvalidConfiguration("libstdc++11 is required for clang.")
+
+        # Validations for dependencies
+        if not self.options["spdlog"].header_only:
+            raise ConanInvalidConfiguration(f"{self.name} requires spdlog:header_only=True to be enabled.")
+
+        if self.options.with_default_http_client and not self.options["cpp-httplib"].with_openssl:
+            raise ConanInvalidConfiguration(f"{self.name} requires cpp-httplib:with_openssl=True when using "
+                                            f"with_default_http_client=True.")
+
+    def package_id(self):
+        if self.options.shared:
+            self.info.shared_library_package_id()
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["ENJINSDK_BUILD_SHARED"] = self.options.shared
+        self._cmake.definitions["ENJINSDK_BUILD_TESTS"] = False
+        self._cmake.configure(source_folder=self._source_subfolder)
```

Comment:
```suggestion
        self._cmake.configure(build_folder=self._build_subfolder)
```
Source folder should point to the Conan build folder on root level, so it will consume the cmake wrapper, which will include the project in source_subfolder.
On the other hand, build_folder can be used separated, so all build cache will be separated from sources